src/libecalc/common/priotity_optimizer.py

```diff
@@ -0,0 +1,81 @@
+import operator
+import typing
+from collections import defaultdict
+from dataclasses import dataclass
+from datetime import datetime
+from functools import reduce
+from typing import Dict, Generic, List, TypeVar
+
+import numpy as np
+from libecalc.common.units import Unit
+from libecalc.common.utils.rates import TimeSeriesBoolean, TimeSeriesInt
+from libecalc.dto import VariablesMap
+
+TResult = TypeVar("TResult")
+TPriority = TypeVar("TPriority")
+
+
+@dataclass
+class PriorityOptimizerResult(Generic[TResult]):
+    priorities_used: TimeSeriesInt
+    priority_results: Dict[datetime, Dict[int, Dict[str, TResult]]]
```

Comment:
Would prefer to have this as a type/class as well to better understand what the different keys and values represent :)